packages/tlschema/src/TLStore.ts

```diff
@@ -44,13 +45,36 @@ export type TLSerializedStore = SerializedStore<TLRecord>
 /** @public */
 export type TLStoreSnapshot = StoreSnapshot<TLRecord>
 
+/** @public */
+export interface TLAssetContext {
+	screenScale: number
+	steppedScreenScale: number
+	dpr: number
+	networkEffectiveType: string | null
+	shouldResolveToOriginal: boolean
+}
+
+/** @public */
+export interface TLAssetStore {
+	upload(asset: TLAsset, file: File): Promise<string>
+	resolve(asset: TLAsset, ctx: TLAssetContext): Promise<string | null> | string | null
+}
+
 /** @public */
 export interface TLStoreProps {
 	defaultName: string
+	assets: TLAssetStore
 }
 
 /** @public */
-export type TLStore = Store<TLRecord, TLStoreProps>
+export class TLStore extends Store<TLRecord, TLStoreProps> {
```

Comment:
I think it'd be fine (and less breaky) to just access the props directly, rather than to create a subclass of `Store` just to wrap them. Some folks might be instantiating `Store` instances without going through `createTLStore` etc, though I'd hope not.